OpenRA.Game/Graphics/Util.cs

```diff
@@ -248,6 +266,41 @@ public static float[] MatrixMultiply(float[] lhs, float[] rhs)
 			return mtx;
 		}
 
+		/// <summary>
+		/// Rotates a quad from a center point around its Z-axis by a given number of radians
+		/// </summary>
+		/// <param name="center">A float3 containing the center point of the quad</param>
+		/// <param name="size"> A float3 containing the X, Y, and Z lengths of the quad</param>
+		/// <param name="rotation">The number of radians to rotate by</param>
+		/// <returns>An array of four vertices representing the rotated quad (top-left, top-right, bottom-right, bottom-left)</returns>
+		#pragma warning disable SA1025 // Code should not contain multiple whitespace in a row
+		public static float3[] RotateQuad(float3 center, float3 size, float rotation)
+		{
+			var angleSin = (float)Math.Sin(-rotation);
+			var angleCos = (float)Math.Cos(-rotation);
+
+			// Rotated offset for +/- x with +/- y
+			var ra = 0.5f * new float3(
+				size.X * angleCos - size.Y * angleSin,
+				size.X * angleSin + size.Y * angleCos,
+				(size.X * angleSin + size.Y * angleCos) * size.Z / size.Y);
+
+			// Rotated offset for +/- x with -/+ y
+			var rb = 0.5f * new float3(
+				size.X * angleCos + size.Y * angleSin,
+				size.X * angleSin - size.Y * angleCos,
+				(size.X * angleSin - size.Y * angleCos) * size.Z / size.Y);
+
+			return new float3[]
```

Comment:
This is a hot code path, so we should be avoiding unnecessary allocations.

Reply:
Ok, but if it is inline then it's not testable. Would assigning the values to a,b,c,d as `out` variables directly work instead?

```c#
public static float3[] RotateQuad(float3 center, float3 size, float rotation, out float3 a, out float3 b, out float3 c, out float3 d)
{
    a = center - ra;
    b = center + rb;
    ...
}
```

Reply:
This is really basic code, so I don't think we should be making it complicated just to satisfy a desire for testing.

Reply:
Ok, I've removed the method and test case, and inlined the code just as it is in your snippet above.